src/app/tools/active_tool.cpp

```diff
@@ -9,6 +9,7 @@
 #include "config.h"
 #endif
 
+#include "app/app.h"
```

Comment:
Oh a little thing I missed at a glance is that we generally include the header file related to the `.cpp` as the first header file (after the `config.h` thing) and then all other files, e.g. to check that `"app/tools/active_tool.h"` includes everything it needs.